### PR TITLE
DM-41711: Upgrade QuantumGraphExecutionReport to handle multiple overlapping graphs

### PR DESCRIPTION
**Pull request by eigerx**
## Checklist

- [x] ran Jenkins
- [x] added a release note for user-visible changes to `doc/changes`

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -3,5 +3,5 @@ numpy
 lsst-daf-butler @ git+https://github.com/lsst/daf_butler@main
 lsst-utils @ git+https://github.com/lsst/utils@main
 lsst-resources @ git+https://github.com/lsst/resources@main
-lsst-pipe-base @ git+https://github.com/lsst/pipe_base@main
-lsst-ctrl-mpexec @ git+https://github.com/lsst/ctrl_mpexec@main
+lsst-pipe-base @ git+https://github.com/lsst/pipe_base@tickets/DM-41711
+lsst-ctrl-mpexec @ git+https://github.com/lsst/ctrl_mpexec@tickets/DM-41711
```
Comment (timj) [inline requirements.txt:7]: You generally want the changes to this file to be in their own commit and not mixed with other code changes so that you can drop the commit on rebase before merging. That's usually easier than editing it and then trying to found the commit to squash into.

**Review by MichelleGower — COMMENTED** [review]
Mostly suggestions about comments, but some code questions.  Merge approved.

**Review by MichelleGower — APPROVED** [review]
Forgot to change to Approve last time and can't figure out how to fix.  So doing it again with merge approved.
